src/ir/item.rs

```diff
@@ -277,11 +278,20 @@ impl Item {
                     TypeKind::Named(ref name, _) => {
                         return name.to_owned();
                     }
-                    _ => {}
-                }
-
-                ty.name().map(ToOwned::to_owned)
-                         .unwrap_or_else(|| format!("_bindgen_ty{}", self.id()))
+                    // We really codegen and use the inner type, so use an empty
+                    // base name so codegen doesn't get confused.
```

Comment:
Thanks, I think I understand much better now. I would phrase it like this:

> We call codegen on the inner type, but we do not want this alias's name to appear in the canonical name just because it is in the type's parent chain, so we use an empty base name.

How is that?

(Thanks for painting this bike shed with me)